src/dodal/beamlines/i18.py

```diff
@@ -0,0 +1,218 @@
+def panda1(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> HDFPanda:
+    return device_instantiation(
+        HDFPanda,
+        "panda1",
+        "-MO-PANDA-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        directory_provider=get_directory_provider(),
+    )
+
+
+@skip_device()
+def xspress3(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> Xspress3:
+    """
+    16 channels Xspress3 detector
+        -EA-XPS-02:CAM:MaxSizeX_RBV
+      also ArraySize
+      also :CONNECTED
+    """
+
+    return device_instantiation(
+        Xspress3,
+        # prefix="-EA-DET-03:",
+        prefix="-EA-XPS-02:",
+        name="Xspress3",
+        num_channels=16,
+        wait=wait_for_connection,
+        fake=fake_with_ophyd_sim,
+    )
+
+
+def dcm(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> DoubleCrystalMonochromator:
+    return device_instantiation(
+        DoubleCrystalMonochromator,
+        "dcm",
+        "",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        bl_prefix=False,
+        motion_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-MO-DCM-01:",
+        temperature_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-DI-DCM-01:",
+        crystal_1_metadata=CrystalMetadata(
+            usage="Bragg",
+            type="silicon",
+            reflection=(1, 1, 1),
+            d_spacing=(3.13475, "nm"),
+        ),
+        crystal_2_metadata=CrystalMetadata(
+            usage="Bragg",
+            type="silicon",
+            reflection=(1, 1, 1),
+            d_spacing=(3.13475, "nm"),
+        ),
+    )
+
+
+@skip_device()
+def i0(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> TetrammDetector:
+    return device_instantiation(
+        TetrammDetector,
+        "i0",
+        "-EA-XBPM-02:",
+        # -DI-XBPM-02:DEV:Firmware
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        type="Cividec Diamond XBPM",
+        directory_provider=get_directory_provider(),
+    )
+
+
+@skip_device()
+def it(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> TetrammDetector:
+    return device_instantiation(
+        TetrammDetector,
+        "it",
+        "-EA-TTRM-02:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        type="PIN Diode",
+        directory_provider=get_directory_provider(),
+    )
+
+
+def vfm(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> FocusingMirror:
+    return device_instantiation(
+        FocusingMirror,
+        "vfm",
+        "-OP-KBM-01:VFM:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+def hfm(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> FocusingMirror:
+    return device_instantiation(
+        FocusingMirror,
+        "hfm",
+        "-OP-KBM-01:HFM:",
```

Comment:
Must: This doesn't connect and from looking on the EDM it seems that the reason is the prefix should actually be `"-OP-HFM-01:"`. Using this instead correctly connects

Reply:
fixed